### PR TITLE
EN-7054: Rest API endpoints throttlers

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
- implemented custom endpoints throttlers in order to be able to limit the maximum number of simultaneous requests (or goroutines) at once.
- removed old implementation for `/transaction/:hash`'s throttler and switched to this approach
- corrected a typo

### CODE REVIEW

File: api/transaction/routes_test.go [hunk 1]
```diff
@@ -132,12 +144,38 @@ func TestGetTransaction_FailsWithWrongFacadeTypeConversion(t *testing.T) {
...
+	req, _ := http.NewRequest("GET", "/transaction/eeee", nil)
+
+	resp := httptest.NewRecorder()
+	ws.ServeHTTP(resp, req)
+
+	transactionResponse := transactionResponse{}
+	loadResponse(resp.Body, &transactionResponse)
+
+	assert.Equal(t, http.StatusTooManyRequests, resp.Code)
+	assert.Equal(t, errors2.ErrTooManyRequests.Error(), transactionResponse.Error)
```
Comment (iulianpascalau) [inline api/transaction/routes_test.go:174]: might change errors2

File: api/transaction/routes_test.go [hunk 2]
```diff
@@ -287,10 +349,40 @@ func TestSendMultipleTransactions_ErrorWithWrongFacade(t *testing.T) {
...
+	req, _ := http.NewRequest("POST", "/transaction/send-multiple", bytes.NewBuffer(jsonBytes))
+
+	resp := httptest.NewRecorder()
+	ws.ServeHTTP(resp, req)
+
+	transactionResponse := sendMultipleTxsResponse{}
+	loadResponse(resp.Body, &transactionResponse)
+
+	assert.Equal(t, http.StatusTooManyRequests, resp.Code)
+	assert.Equal(t, errors2.ErrTooManyRequests.Error(), transactionResponse.Error)
```
Comment (iulianpascalau) [inline api/transaction/routes_test.go:383]: errors2

File: cmd/node/config/config.toml [hunk 3]
```diff
@@ -383,6 +383,10 @@
         SameSourceRequests = 10000
         # SameSourceResetIntervalInSec time frame between counter reset, in seconds
         SameSourceResetIntervalInSec = 1
+        # EndpointsThrottlers represents a map for maximum simultaneous go routines for an endpoint
+        EndpointsThrottlers = [{ Endpoint = "/transaction/get", MaxNumGoRoutines = 10 },
```
Comment (iulianpascalau) [inline cmd/node/config/config.toml:387]: Might lower these values to 2

File: api/transaction/routes.go [hunk 4]
```diff
@@ -87,6 +89,24 @@ func SendTransaction(c *gin.Context) {
 		return
 	}
 
+	endpointThrottler, ok := ef.GetThrottlerForEndpoint("/transaction/send")
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply api/transaction/routes.go:92]: done

File: api/transaction/routes_test.go (hunk 1 above)
Comment (bogdan-rosianu) [reply api/transaction/routes_test.go:174]: done

File: api/transaction/routes_test.go (hunk 2 above)
Comment (bogdan-rosianu) [reply api/transaction/routes_test.go:383]: done

File: cmd/node/config/config.toml (hunk 3 above)
Comment (bogdan-rosianu) [reply cmd/node/config/config.toml:387]: changed the `send` endpoint to support `2` but left the "get" endpoint value to `10` as it was before

Comment (iulianpascalau) [reply cmd/node/config/config.toml]: ok :+1:

File: facade/nodeFacade.go [hunk 5]
```diff
@@ -92,16 +94,37 @@ func NewNodeFacade(arg ArgNodeFacade) (*nodeFacade, error) {
 		return nil, fmt.Errorf("%w, SameSourceResetIntervalInSec should not be 0", ErrInvalidValue)
 	}
 
+	throtlersMap := computeEndpointsNumGoRoutinesThrottlers(arg.WsAntifloodConfig)
```
Comment (sasurobert) [inline facade/nodeFacade.go:97]: throttlersMap

Comment (bogdan-rosianu) [reply facade/nodeFacade.go]: fixed

**Pull request merged**